File: backend/src/main/java/edu/cibertec/stockwarehouse/tipousuario/domain/model/TipoUsuario.java
package edu.cibertec.stockwarehouse.tipousuario.domain.model;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Table(name = "tb_tipo_usuario")
public class TipoUsuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_tipo_usuario")
    private long idTipoUsuario;

    @Column(name = "nombre_tipo_usuario", length = 50, nullable = false)
    private String nombreTipoUsuario;

    @Column(name = "estado")
    private int estado;

    public String NombreEstado() {
        String nombreEstado = null;
        switch (estado) {
            case 0:
                nombreEstado = "Activo";
                break;
            case 1:
                nombreEstado = "Anulado";
                break;
            default:
                nombreEstado = "";
        }
        return nombreEstado;
    }
}

File: frontend/src/app/data/schema/tipousuario.ts
export class TipoUsuario{
    
    idTipoUsuario: number;
    nombreTipoUsuario: string;
    estado: number;

    constructor(){
        this.idTipoUsuario=0;
        this.nombreTipoUsuario="";
        this.estado=0;
    }

    NombreEstado(estado:number):string {

        let nombreEstado:string = "";

       switch (estado) {
           case 0:
               nombreEstado = "Generado";
               break;
           case 1:
               nombreEstado = "Recepcionado";
               break;
           case 2:
               nombreEstado = "Anulado";
               break;
           default:
               nombreEstado = "Otros";
       }

       return nombreEstado;

   }
}